xmtp_mls/src/client.rs

```diff
@@ -344,28 +344,38 @@ where
 
     // set the consent record in the database
     // if the consent record is an address also set the inboxId
-    pub async fn set_consent_state(
+    pub async fn set_consent_states(
         &self,
-        state: ConsentState,
-        entity_type: ConsentType,
-        entity: String,
+        mut records: Vec<StoredConsentRecord>,
     ) -> Result<(), ClientError> {
         let conn = self.store().conn()?;
-        conn.insert_or_replace_consent_record(StoredConsentRecord::new(
-            entity_type,
-            state,
-            entity.clone(),
-        ))?;
 
-        if entity_type == ConsentType::Address {
-            if let Some(inbox_id) = self.find_inbox_id_from_address(entity.clone()).await? {
-                conn.insert_or_replace_consent_record(StoredConsentRecord::new(
-                    ConsentType::InboxId,
-                    state,
-                    inbox_id,
-                ))?;
+        // Collect new records to add after the loop
+        let mut new_records = Vec::new();
+
+        // Iterate over the records and check if the entity is an address
+        for record in &records {
+            if record.entity_type == ConsentType::Address {
+                // Find the inbox ID associated with the address asynchronously
+                if let Some(inbox_id) = self
+                    .find_inbox_id_from_address(record.entity.clone())
```

Comment:
Would be nice to batch this call, since it goes to the network

Reply:
Good callout! I just added the batched API call.